frontend/cypress/integration/talentsearch/footer.spec.js

```diff
@@ -25,7 +25,7 @@ describe('Footer', () => {
 
     it('links to Canada.ca', () => {
       cy.findByRole('contentinfo').within(() => {
-        cy.findByText('Canada.ca').should('have.attr', 'href', 'https://www.canada.ca/en.html')
+        cy.findAllByRole('link', { name: /canada.ca/i }).eq(0).should('have.attr', 'href', 'https://www.canada.ca/en.html')
```

Comment:
Why not findByRole? I'm not sure this is how a test should be written, as it's very non-specific when it should be finding exactly  one (and I'm wondering if it's actually saying "there should be an array of zero results", and that's why it's passing...)

Reply:
Ah, I'm seeing that it's for getting a specific element in array, but still feels better to actually target the exact text and avoid findAllByRole when possible. This seems to be less clear, for the sake of using `name: /foo/i` syntax?

Reply:
I made this change to specifically target `<a>` rather than any element with the text. Though, we have two links with that text (may have been a change?) which is the reason for `findAllBy*`. However, I have no idea to check if both have the proper `href` so I changed to just check the first one. 

Any assistance on how to check both would be appreciated. :)

Reply:
I think I figured it out in [test both canada footer links](https://github.com/GCTC-NTGC/gc-digital-talent/pull/4049/commits/27edd925ae4ceb4be47a832565eaaa63a89fe4fd)

At least, it is passing locally :P